### PR TITLE
Bug 1862996 : Add OLM Dependency Resolution Metric

### PR DESCRIPTION
**Pull request by awgreene**
OLM recently introduced a new SAT-based dependency resolver. This
commit introduces a metric that allows OLM to track worst-case performances.

* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: Documentation/data-collection.md [hunk 1]
```diff
@@ -134,6 +134,8 @@ data:
     # subscription_sync_total is the number of times an OLM operator
     # Subscription has been synced, labelled by name and installed csv
     - '{__name__="subscription_sync_total"}'
+    # olm_resolution_duration_seconds is the duration of a dependency resolution attempt
```
Comment (s-urbaniak) [inline Documentation/data-collection.md:137]: nit: i believe this is missing an owning team. I assume this is `@openshift/openshift-team-olm`?

**Comment by s-urbaniak** [comment]
As per OOB conversation with @smarterclayton and this is approved and we assume a constant amount of 6 series with this metric.

I have one nit, else lgtm.

File: Documentation/data-collection.md (hunk 1 above)
Comment (awgreene) [reply Documentation/data-collection.md:137]: I was unaware that the OLM team had a tag, where was @openshift/openshift-team-olm generated ?

Comment (awgreene) [reply Documentation/data-collection.md:137]: Added in the latest version of the release.

**Comment by awgreene** [comment]
```
{"component":"entrypoint","file":"prow/entrypoint/run.go:168","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Entrypoint received interrupt: terminated","severity":"error","time":"2020-07-30T17:46:00Z"}
```
/retest

**Comment by awgreene** [comment]
@s-urbaniak can we get eyes on this again? I assume we need a bz now?

**Pull request merged**